pkg/lighthouse/crds.go

```diff
@@ -0,0 +1,46 @@
+package lighthouse
+
+import (
+	"fmt"
+
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+// Ensure ensures that the required resources are deployed on the target system
+// The resources handled here are the lighthouse CRDs: MultiClusterService
+func Ensure(config *rest.Config) error {
```

Comment:
Doesn't _pkg/subctl/operator/lighthouse/crds/ensure.go_ also ensure/install the `MultiClusterService` CRD?  If so what is the file for and why does it do it differently?

Reply:
This for the deployBroker, it takes a path different from  join where this pkg/subctl/operator/lighthouse/crds/ensure.go will be executed.

Reply:
right but why 2 different code paths to do the create? Why can't they reuse the same code to deploy the same CRD? This creates the CRD in code while the other generates it from embedded yaml with slight differences. I'd suggest one call into the other to create the CRD or at last share the same `GetMcsCRD` function which either creates in code or generates from yaml.

Reply:
I have made getMcsCrd a common function which uses the yaml.